kernel/src/fs/procfs/pid/stat.rs

```diff
@@ -68,36 +76,113 @@ use crate::{
 /// - env_start        : Start address of environment variables.
 /// - env_end          : End address of environment variables.
 /// - exit_code        : Process exit code as returned by waitpid(2).
-pub struct StatFileOps(Arc<Process>);
+pub struct StatFileOps {
+    process_ref: Arc<Process>,
+    thread_ref: Arc<Thread>,
+    is_pid_stat: bool,
+}
 
 impl StatFileOps {
-    pub fn new_inode(process_ref: Arc<Process>, parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
-        ProcFileBuilder::new(Self(process_ref))
-            .parent(parent)
-            .build()
-            .unwrap()
+    pub fn new_inode(
+        process_ref: Arc<Process>,
+        thread_ref: Arc<Thread>,
+        is_pid_stat: bool,
+        parent: Weak<dyn Inode>,
+    ) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self {
+            process_ref,
+            thread_ref,
+            is_pid_stat,
+        })
+        .parent(parent)
+        .build()
+        .unwrap()
     }
 }
 
 impl FileOps for StatFileOps {
     fn data(&self) -> Result<Vec<u8>> {
-        let process = &self.0;
+        let process = &self.process_ref;
+        let thread = &self.thread_ref;
+        let posix_thread = thread.as_posix_thread().unwrap();
 
```

Comment:
The PR description has a well-written summary to explain the Linux implementation. I think they can be useful implementation notes here. Something like:
```rust
        // According to the Linux implementation, a process's `/proc/<pid>/stat` should be ...
        //
        // Reference: <http://...>

        let pid = posix_thread.tid();
        // ...
```

The same applies for the case in `pid/status.rs`.